### PR TITLE
[JENKINS-32133] Styling pipeline reference

### PR DESCRIPTION
**Pull request by gusreiber**

### CODE REVIEW

**Comment by gusreiber** [comment]
@reviewbybees

**Comment by recena** [comment]
@gusreiber Is there any issue for that?

**Comment by amuniz** [comment]
Please, add a JIRA issue reference in the title and a link in the description.

File: cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/css/workflow.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.steps-box p {margin:0; padding-bottom:1em}
```
Comment (amuniz) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/css/workflow.css]: This file is really difficult to read. Could you format it in a better way?

```
.steps-box p {
    margin:0; 
    padding-bottom:1em
}
```
That is much better than the one-liner style.

File: cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy [hunk 2]
```diff
@@ -12,47 +12,64 @@ Snippetizer snippetizer = my;
...
 
-
-h1(_("Steps"))
-for (StepDescriptor d : snippetizer.getStepDescriptors(false)) {
-    generateStepHelp(d);
+st.adjunct(includes: 'org.jenkinsci.plugins.workflow.cps.Snippetizer.css.workflow')
+
+h1(_("DSL Reference"))
+div(class:'steps-box basic'){
+	h2(_("Steps"))
```
Comment (amuniz) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy]: This file was previously formatted with 2 spaces and no tabs, follow the previous format please.

File: cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy [hunk 3]
```diff
@@ -12,47 +12,64 @@ Snippetizer snippetizer = my;
...
+			}
+		}
+	}
 }
 
-
 def generateStepHelp(StepDescriptor d) throws Exception {
     return {
-        h2 {
+    	dt(class:'step-title'){
```
Comment (amuniz) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy]: Two spaces instead of tabs.

**Comment by abayer** [comment]
JIRA is probably https://issues.jenkins-ci.org/browse/JENKINS-32133

**Comment by gusreiber** [comment]
I'll reformat the files.

**Comment by jglick** [comment]
A lot of reindentation here; [this](https://github.com/jenkinsci/workflow-plugin/pull/320/files?w=1) is easier to review.

File: cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy [hunk 4]
```diff
@@ -148,4 +184,5 @@ def describeType(DescribableHelper.ParameterType type, int headerLevel) throws E
             assert false: type;
         }
     }.call()
-}
\ No newline at end of file
+}
+st.adjunct(includes: 'org.jenkinsci.plugins.workflow.cps.Snippetizer.js.workflow')
```
Comment (jglick) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy]: Looks wrong; what includes the CSS? I think you meant to use an adjunct name like `org.jenkinsci.plugins.workflow.cps.Snippetizer.reference` with resources named `org/jenkinsci/plugins/workflow/cps/Snippetizer/reference.js` and `org/jenkinsci/plugins/workflow/cps/Snippetizer/reference.css`.

Comment (jglick) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy]: Oh I see, you have _another_ adjunct at line 15. You only need one to include all associated resources. :bug:

**Comment by abayer** [comment]
Does it make sense to not expand all the steps initially?

**Comment by abayer** [comment]
The environment variables aren't nested properly - 
image

**Comment by abayer** [comment]
And as a result of that, variables after `env` don't render right - 
image

File: cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy (hunk 3 above)
Comment (abayer) [inline cps/src/main/resources/org/jenkinsci/plugins/workflow/cps/Snippetizer/dslReferenceContent.groovy]: The step titles are now the same size as the parameter titles, and feel too big in general. Might make sense to shrink both, and have different sizes for nested titles again?

**Comment by gusreiber** [comment]
Added the following fixes:
-- put spaces back for tabs
-- formatted css with spaces and line breaks
-- fixed misuse of adjunct tag
-- fixed tag nesting rule error in EnvActionImpl...help.jelly (DL tag needed for DT enteries)

**Comment by gusreiber** [comment]
screen shot 2016-01-25 at 8 19 29 am

**Comment by gusreiber** [comment]
@abayer I don't think it makes sense to minimize the items initially, as that would cripple the browser's find string capability.

**Comment by jglick** [comment]
:bee:, just need to restart PR builder

**Comment by gusreiber** [comment]
I can also build a ToC out of the steps, which given the length of this file, might be helpful, but I am hesitant to pile too much goop into this otherwise straightforward display.

**Comment by amuniz** [comment]
> I can also build a ToC

Better than a ToC would be a search field filtering the DOM (removing non-matches). But that's clearly out-of-scope for now.

**Comment by gusreiber** [comment]
@abayer Can I get a second bee from you?

**Comment by abayer** [comment]
:bee:

**Comment by gusreiber** [comment]
It looks like I don't have permission to merge this guy.

**Comment by abayer** [comment]
@reviewbybees done - @jglick, I leave it to you to merge. =)

**Pull request merged**